en/middleware/micrortps.md

```diff
@@ -30,58 +32,84 @@ The main elements of the architecture are the client and agent processes shown i
 
 <span></span>
-> **Tip** The bridge code can also be [manually generated](micrortps_manual_code_generation.md). Most users will not need to do so, but the linked topic provides a more detailed overview of the build process and can be useful for troubleshooting.
+> **Tip** The bridge code can also be [manually generated](micrortps_manual_code_generation.md). Most users will not need to do
+so, but the linked topic provides a more detailed overview of the build process and can be useful for troubleshooting.
 
+### Code generation for ROS2/ROS applications - `px4_ros_com`
 
-## Supported uORB messages
-
-The generated bridge code will enable a specified subset of uORB topics to be published/subscribed via RTPS.
+The [`px4_ros_com`](https://github.com/PX4/px4_ros_com) package allows the developer to generate all the required ROS2/ROS message headers and sources, but also the agent application which will bridge the client with the RTPS middleware on the ROS2 side. The package is basically composed by two separate branches, where the `master` branch generates all the required ROS2 messages and IDL files, besides some listener and advertiser example nodes, in order to bridge PX4 with ROS2 nodes, and a `ros1` branch, which similarly to the `master` branch, generates the ROS(1) message headers and source files, besides the ROS nodes examples, that allow, by using the `ros1_bridge`, share data between PX4 and ROS.
 
-For *automatic code generation* (via the normal PX4 firmware build process) this set must be listed in the **/path/to/PX4/Firmware/src/modules/micrortps_bridge/CMakeLists.txt** file.
+The `px4_ros_com` package build structure is structured in a way that, when one builds the package, it automatically generates all the required components of the bridge that allow the developer to use the generated messages on its own nodes, being it on the ROS2 or in the ROS side.
 
-```cmake
-set(config_rtps_send_topics
-  sensor_combined
-   # Add new topic...
-   )
+## Supported uORB messages
 
-set(config_rtps_receive_topics
-   sensor_baro
-   # Add new topic...
-   )
+The generated bridge code will enable a specified subset of uORB topics to be published/subscribed via RTPS. This applicable on both ROS or non-ROS applications.
+
+For *automatic code generation* (via the normal PX4 Firmware build process or, in the case of the `px4_ros_com`, through the build process of the package), there's a yaml definition file on the PX4 Firmware directory called `uorb_rtps_message_ids.yaml`, found in **/path/to/PX4/Firmware/msg/tools/** dir, where all the uORB messages (RTPS) ID's are set and where is also set which messages should be set to be sent, received, or both, in the RTPS stream.
+
+> **Caution** All new uORB messages which one adds to the Firmware side should have a respective RTPS ID set when one wants the data to be published and/or subscribed in the RTPS stream.
```

Comment:
Are you saying that the uORB definition in firmware source code must include an RTPS ID field? If so, we should show an example. 
Or are you saying that an ID needs to be in the YAML as shown below.
Or are you saying both - you define a field in firmware and have to match it up here in the YAML?

What is the strategy for allocating ids?

Reply:
You just need to add the RTPS ID to the respecrive msg/topic on the `uorb_rtps_message_ids.yaml` file. And that's it.